scipy/stats/_continuous_distns.py

```diff
@@ -3554,6 +3559,65 @@ def fit(self, data, *args, **kwds):
             fshape_s = fshape_n / fscale
         return fshape_s, floc, fscale
 
+    def _ppf_isf(self, q, mu, upper):
+        """ calculate ppf or isf depending on the value of `upper`.
+        If upper is True, then compute isf, else ppf. This implementation is
+        based on Gyner & Smyth (2016) and uses newton's method with carefully
+        selected starting points that guarantee convergence regardless of the
+        probability."""
+
+        def f(a, b, c, d, e):
+            """approximates (x0 - norm_cdf) / pdf when abs(x0 - norm_cdf) < 1e-5"""
+            return a * np.exp(c + np.log1p(-0.5 * a) - d)
+
+        def f2(a, b, c, d, e):
+            """computes (x0 - norm_cdf) / pdf"""
+            return b * np.exp(-d) - np.exp(e - d)
+
+        def f_small_q(q, mu, k):
+            """select the start values for the newton using the quantiles of
+            the gamma (if right tail is required) or normal distribution"""
+            if upper:
+                return gamma.isf(q, 1 / mu, scale=mu * mu)
+            return mu / (_norm_ppf(q) ** 2),
+
+        k = 1.5 * mu
+        # get starting points
+        x0 = _lazywhere(
+            q < 1e-5,
+            (q, mu, k),
+            f_small_q,
+            f2=lambda q, mu, k: _lazywhere(
+                mu > 100,
```

Comment:
Not sure about the appropriateness of 100 here as a threshold? I picked arbitrarily. Maybe a much bigger value would be better? I dont know.

Reply:
There are also some magical numbers in the true branch. Are they also coming from the article or just fractions spelled out?

Reply:
Those are not magical, they are fractions from the taylor expansion on page 2 of the article.